.github/workflows/publish-docker.yml

```diff
@@ -1,16 +1,31 @@
 name: Docker Build + Publish
 
 on:
-  workflow_dispatch:
+  workflow_call:
     inputs:
-      docker_tag:
-        description: "tag to be used for the docker image"
-        type: string
-        required: true
       push_to_repository:
-        description: "whether to push out the docker image: no (default) or yes"
+        description: "whether to push out the docker image: false (default) or true"
+        type: boolean
+        default: false
+      gitref:
+        description: "git commit hash or branch name"
+        type: string
+        default: 'main'
+      build_type:
```

Comment:
why are we making a new name for a variable that is already present? `build_type` seems to be the same as `wf_category`.

Reply:
Good idea, updated.